prefixfilter.go

```diff
@@ -0,0 +1,48 @@
+package complete
+
+import (
+	"strings"
+)
+
+//PrefixFilter filters a predictor's options based on the prefix
+type PrefixFilter interface {
+	FilterPrefix(str, prefix string) bool
+}
+
+//PrefixFilteringPredictor is a Predictor that also implements PrefixFilter
+type PrefixFilteringPredictor struct {
+	Predictor        Predictor
+	PrefixFilterFunc func(s, prefix string) bool
+}
+
+func (p *PrefixFilteringPredictor) Predict(a Args) []string {
+	if p.Predictor == nil {
+		return []string{}
+	}
+	return p.Predictor.Predict(a)
+}
+
+func (p *PrefixFilteringPredictor) FilterPrefix(str, prefix string) bool {
+	if p.PrefixFilterFunc == nil {
+		return DefaultPrefixFilter(str, prefix)
+	}
+	return p.PrefixFilterFunc(str, prefix)
+}
+
+//DefaultPrefixFilter is the PrefixFilter used when none is set
+func DefaultPrefixFilter(s, prefix string) bool {
```

Comment:
I think this could be private, as I don't see anyone use it - they will just use a regular predictor.